### PR TITLE
[PT-2.0] SM training release

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
1. implement the SM validation and benchmarking tests for inductor
2. refactor and fix mnist distributed training tests for pytorch.
3. incorporate conda AWS-PyTorch, changed cuda linkage to conda env.
4. optimize some docker definitions as an attempt to reduce container size
5. apply Rubik NCCL hang issue fix in training toolkit fork, apply Rubik/horovod segfault fix in an smdebug upgrade,

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/sagemaker_tests/pytorch/training/resources/mnist/pytorchddp_throughput_mnist.py [hunk 1]
```diff
@@ -160,6 +160,10 @@ def main():
                         type=str,
                         default='/tmp/data',
                         help='Path for downloading the MNIST dataset')
+    parser.add_argument('--inductor', 
+                        type=int, 
+                        default=0,
+                        help='trcomp with inductor')
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/mnist/pytorchddp_throughput_mnist.py:166]: This is unrelated to trcomp. Use "pytorch with inductor"

File: test/sagemaker_tests/pytorch/training/resources/mnist/smmodelparallel_pt_mnist.py [hunk 2]
```diff
@@ -226,6 +226,8 @@ def get_parser():
     parser.add_argument("--data-dir", type=str, default=None)
     parser.add_argument("--ddp", type=int, default=0)
     parser.add_argument('--mp_parameters', type=str, default='')
+    parser.add_argument('--inductor', type=int, default=0,
+                        help='trcomp with inductor')
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/mnist/smmodelparallel_pt_mnist.py:230]: This is unrelated to trcomp. Use "pytorch with inductor"

File: test/sagemaker_tests/pytorch/training/resources/resnet18/main.py [hunk 3]
```diff
@@ -93,6 +93,8 @@
                          'N processes per node, which has N GPUs. This is the '
                          'fastest way to use PyTorch for either single node or '
                          'multi node data parallel training')
+parser.add_argument('--inductor', type=int, default=0,
+                        help='trcomp with inductor')
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/resnet18/main.py:97]: This is unrelated to trcomp. Use "pytorch with inductor"

File: test/sagemaker_tests/pytorch/training/resources/gpt2/train_gpt_simple.py [hunk 4]
```diff
@@ -820,6 +820,8 @@ def parse_args():
     parser.add_argument(
         "--smp_version", type=int, default=110, help="Enable memory profile"
     )
+    parser.add_argument('--inductor', type=int, default=0,
+                        help='trcomp with inductor')
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/gpt2/train_gpt_simple.py:824]: This is unrelated to trcomp. Use "pytorch with inductor"

File: test/sagemaker_tests/pytorch/training/resources/dgl-gcn/train.py [hunk 5]
```diff
@@ -58,6 +58,8 @@ def train(g, features, labels, masks, model):
     parser = argparse.ArgumentParser()
     parser.add_argument("--dataset", type=str, default="cora",
                         help="Dataset name ('cora', 'citeseer', 'pubmed').")
+    parser.add_argument('--inductor', type=int, default=0,
+                        help='trcomp with inductor')
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/dgl-gcn/train.py:62]: This is unrelated to trcomp. Use "pytorch with inductor"

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
+# Copyright 2018-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py:1]: Check framework version and skip inductor tests if PT < 2.0

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_mnist_inductor.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
+# Copyright 2018-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_mnist_inductor.py:1]: Check framework version and skip inductor tests if PT < 2.0

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations_inductor.py [hunk 8]
```diff
@@ -0,0 +1,618 @@
+# Copyright 2018-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations_inductor.py:1]: Check framework version and skip inductor tests if PT < 2.0

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pt_s3_plugin_inductor.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
+# Copyright 2018-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pt_s3_plugin_inductor.py:1]: I believe S3 plugin as a feature has been deprecated. Confirm with Shibo and remove file if unnecessary.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pytorchddp_inductor.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
+# Copyright 2018-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pytorchddp_inductor.py:2]: Check framework version and skip inductor tests if PT < 2.0

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    invoke_pytorch_helper_function(ecr_image, sagemaker_regions, _test_dgl_training, function_args)
+
+
+@pytest.mark.integration("dgl")
+@pytest.mark.processor("gpu")
+@pytest.mark.model("gcn")
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+def test_dgl_gcn_training_gpu(ecr_image, sagemaker_regions, instance_type):
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py:58]: Ensure instance coverage is p3.2xl, g5.4xl, g4dn.4xl

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    
+    # TODO: Remove when DGL gpu test on ecs get fixed
+    if Version(image_framework_version) in SpecifierSet("==1.10.*") and image_cuda_version == "cu113":
+        pytest.skip("ecs test for DGL gpu fails for pt 1.10")
+
+    if Version(image_framework_version) == Version("1.6") and image_cuda_version == "cu110":
+        pytest.skip("DGL does not support CUDA 11 for PyTorch 1.6")
+
+    instance_type = instance_type or "ml.p2.xlarge"
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py:69]: p2 and g3 instances are not supported with Inductor.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    function_args = {
+        "instance_type": instance_type,
+    }
+
+    if Version(image_framework_version) in SpecifierSet(">=1.11.0"):
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py:74]: Inductor only exists in PT>2.0 so remove any irrelevant code.

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_performance_inductor.py [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+        pytorch = PyTorch(
+            entry_point=inductor_script,
+            role='SageMakerRole',
+            instance_count=1,
+            instance_type=instance_type,
+            sagemaker_session=sagemaker_session,
+            image_uri=ecr_image,
+            framework_version=framework_version,
+        )
+        training_input = pytorch.sagemaker_session.upload_data(path=training_dir, key_prefix='pytorch/inductor')
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_performance_inductor.py:53]: Torchdynamo benchmarks use fake data. This should not be necessary

File: test/sagemaker_tests/pytorch/training/resources/inductor/run_pytorch_inductor.sh [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+
+set -e
+cd $BIN_DIR
+# tmp fix for numpy deprecations, PR raised: https://github.com/HewlettPackard/dlcookbook-dlbs/pull/19
+git clone --quiet --single-branch --depth 1 --branch v2.0.0 https://github.com/pytorch/pytorch.git pytorch
+
+cd pytorch
+
+pip install -U numpy
+pip install deepspeed==0.8.2
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/inductor/run_pytorch_inductor.sh:28]: Remove deepspeed.

File: test/sagemaker_tests/pytorch/training/resources/inductor/run_pytorch_inductor.sh [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+    echo "Training Huggingface Complete using PyTorch Inductor."
+else
+    echo "Training Huggingface Failed using PyTorch Inductor."
+    cat $TRAINING_LOG
+    exit 1
+fi
+
+TRAINING_LOG=${LOG_DIR}/pytorch_inductor_timm_benchmark.log
+
+python benchmarks/dynamo/runner.py --suites=timm_models --training --dtypes=amp --compilers=inductor --output-dir=timm_logs --extra-args='--output-directory=./' > $TRAINING_LOG 2>&1
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/inductor/run_pytorch_inductor.sh:49]: Set output directory to /opt/ml/output/data for the artifacts to be automatically picked up by sagemaker - https://docs.aws.amazon.com/sagemaker/latest/dg/model-train-storage.html

Upload logs to the same location.

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_performance_inductor.py [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+from ...integration import training_dir, DEFAULT_TIMEOUT, inductor_script
+from ...integration.sagemaker.timeout import timeout
+from .... import invoke_pytorch_helper_function
+
+PR_PARAMS = {"instance_type": ["ml.p3.2xlarge", "ml.g5.4xlarge", "ml.g4dn.4xlarge"]}
+
+@pytest.mark.processor("gpu")
+@pytest.mark.skip_cpu
+def test_inductor_performance_gpu(framework_version, ecr_image, sagemaker_regions, instance_type):
+    instance_type = "ml.p3.2xlarge"
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_performance_inductor.py:34]: Remove the hard coding here. We have already set instance type from PR_PARAMS

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py [hunk 18]
```diff
@@ -55,40 +54,19 @@ def test_dgl_gcn_training_cpu(ecr_image, sagemaker_regions, instance_type):
 @pytest.mark.model("gcn")
 @pytest.mark.skip_cpu
 @pytest.mark.skip_py2_containers
+@pytest.mark.skip_inductor_test
 def test_dgl_gcn_training_gpu(ecr_image, sagemaker_regions, instance_type):
     _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
     image_cuda_version = get_cuda_version_from_tag(ecr_image)
     
     # TODO: Remove when DGL gpu test on ecs get fixed
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py:62]: Remove comment.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_performance_inductor.py [hunk 19]
```diff
@@ -22,36 +22,68 @@
...
-            entry_point=inductor_script,
+            entry_point=f"run_inductor_{suites}.sh",
+            source_dir=inductor_path,
             role='SageMakerRole',
             instance_count=1,
             instance_type=instance_type,
             sagemaker_session=sagemaker_session,
             image_uri=ecr_image,
             framework_version=framework_version,
+            output_path=f"s3://sagemaker-inductor-test-{suites}",
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_performance_inductor.py:84]: Ensure this bucket exists in the CI account. Please create if it does not exist.

File: test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_huggingface.sh [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+
+set -e
+cd $BIN_DIR
+# tmp fix for numpy deprecations, PR raised: https://github.com/HewlettPackard/dlcookbook-dlbs/pull/19
+git clone --quiet --single-branch --depth 1 --branch v2.0.0 https://github.com/pytorch/pytorch.git pytorch
+
+cd pytorch
+
+pip install -U numpy
+pip install deepspeed==0.8.2
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_huggingface.sh:28]: Remove deepspeed

File: test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_huggingface.sh [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+pip install gitpython
+pip install tabulate==0.9.0
+
+TRAINING_LOG=${LOG_DIR}/pytorch_inductor_huggingface_benchmark.log
+
+python benchmarks/dynamo/runner.py --suites=huggingface --training --dtypes=amp --compilers=inductor --output-dir=huggingface_logs --extra-args='--output-directory=./' > $TRAINING_LOG 2>&1
+
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_huggingface.sh:35]: Upload the benchmarks results to cloudwatch metrics for dashboarding

File: test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_timm.sh [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+set -e
+
+if [ ${RETURN_VAL} -eq 0 ]; then
+    echo "Training Timm Complete using PyTorch Inductor."
+else
+    echo "Training Timm Failed using PyTorch Inductor."
+    cat $TRAINING_LOG
+    exit 1
+fi
+
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_timm.sh:51]: Upload the benchmarks results to cloudwatch metrics for dashboarding

File: test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_torchbench.sh [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+set -e
+
+if [ ${RETURN_VAL} -eq 0 ]; then
+    echo "Training Torchbench Complete using PyTorch Inductor."
+else
+    echo "Training Torchbench Failed using PyTorch Inductor."
+    cat $TRAINING_LOG
+    exit 1
+fi
+
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/resources/inductor/run_inductor_torchbench.sh:63]: Upload the benchmarks results to cloudwatch metrics for dashboarding

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

File: pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu [hunk 24]
```diff
@@ -233,8 +231,9 @@ RUN mkdir /tmp/efa-ofi-nccl \
  && ./autogen.sh \
  && ./configure --with-libfabric=/opt/amazon/efa \
   --with-mpi=/opt/amazon/openmpi \
-  --with-cuda=/usr/local/cuda \
-  --with-nccl=/usr/local --prefix=/usr/local \
```
Comment (roywei) [inline pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu:237]: there is no need to install ofi, as it comes with conda install

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_inductor_helper.py [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+from botocore.exceptions import ClientError
+import datetime
+import pandas as pd
+import os
+import argparse
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+AWS_DEFAULT_REGION = "us-west-2"
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_inductor_helper.py:12]: use existing environment variable `os.environ['AWS_DEFAULT_REGION']`

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

**Comment by ShiboXing** [comment]
marked failing inductor tests as xfail, as the failures are either build or upstream torch.compile issues. Will report to conda team/upstream.

**Comment by ShiboXing** [comment]
SM test:
standard: 8b6c17b
efa: 1dcda01
rc: dbaa8a4

File: test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_huggingface.sh [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+mkdir -p ${LOG_DIR}
+
+python -c "import sys;
+assert sys.version_info < (3, 7);"
+PYTHON_VERSION_CHECK=`echo $?`
+
+set -e
+cd $BIN_DIR
+# tmp fix for numpy deprecations, PR raised: https://github.com/HewlettPackard/dlcookbook-dlbs/pull/19
+git clone --quiet --single-branch --depth 1 --branch v2.0.0 https://github.com/pytorch/pytorch.git pytorch
```
Comment (xncqr) [inline test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_huggingface.sh:24]: This should likely be parameterized and like using an ENV variable so we are not hardcoding versions in multiple areas.

File: test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_timm.sh [hunk 27]
```diff
@@ -0,0 +1,47 @@
...
+mkdir -p ${LOG_DIR}
+
+python -c "import sys;
+assert sys.version_info < (3, 7);"
+PYTHON_VERSION_CHECK=`echo $?`
+
+set -e
+cd $BIN_DIR
+# tmp fix for numpy deprecations, PR raised: https://github.com/HewlettPackard/dlcookbook-dlbs/pull/19
+git clone --quiet --single-branch --depth 1 --branch v2.0.0 https://github.com/pytorch/pytorch.git pytorch
```
Comment (xncqr) [inline test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_timm.sh:23]: Same as test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_huggingface.sh. The version should not be hardcoded

File: test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_torchbench.sh [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+mkdir -p ${LOG_DIR}
+
+python -c "import sys;
+assert sys.version_info < (3, 7);"
+PYTHON_VERSION_CHECK=`echo $?`
+
+set -e
+cd $BIN_DIR
+# tmp fix for numpy deprecations, PR raised: https://github.com/HewlettPackard/dlcookbook-dlbs/pull/19
+git clone --quiet --single-branch --depth 1 --branch v2.0.0 https://github.com/pytorch/pytorch.git pytorch
```
Comment (xncqr) [inline test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_torchbench.sh:23]: The version should not be hardcoded, this is the third file. When we go to 2.0.1 or any other version, all these would need to be updated each time.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py (hunk 11 above)
Comment (Lokiiiiii) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl_inductor.py:57]: Why is this marked as resolved ? I don't see coverage for g5 and g4dn.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pytorchddp_inductor.py [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+
+
+@pytest.mark.processor("gpu")
+@pytest.mark.model("N/A")
+@pytest.mark.multinode(2)
+@pytest.mark.integration("pytorchddp")
+@pytest.mark.parametrize('efa_instance_type', get_efa_test_instance_type(default=["ml.p4d.24xlarge"]), indirect=True)
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+@pytest.mark.skip_trcomp_containers
```
Comment (Lokiiiiii) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pytorchddp_inductor.py:45]: This is unrelated to trcomp

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

File: test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_huggingface.sh (hunk 26 above)
Comment (ShiboXing) [reply test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_huggingface.sh:24]: Will improve this in a new PR for inductor benchmarking

File: test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_timm.sh (hunk 27 above)
Comment (ShiboXing) [reply test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_timm.sh:23]: Will improve this in a new PR for inductor benchmarking

File: test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_torchbench.sh (hunk 28 above)
Comment (ShiboXing) [reply test/dlc_tests/benchmark/sagemaker/pytorch/training/resources/inductor/run_inductor_torchbench.sh:23]: Will improve this in a new PR for inductor benchmarking

**Review by Lokiiiiii — APPROVED** [review]

**Review by junpuf — DISMISSED** [review]
LGTM!

**Review by junpuf — DISMISSED** [review]

**Review by yl-to — APPROVED** [review]
smdebug version LGTM, approving.

**Review by junpuf — APPROVED** [review]
